docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -201,44 +201,117 @@ The ShootState synchronization controller will become part of the gardenlet. It
+
+1. The reconciliation flow ("restore" phase) is triggered in the destination seed without first executing the migration flow in the source seed (or perhaps it was executed, but it failed, and its state is currently unknown).
+
+2. An additional "source" backup entry referencing the source seed backup bucket is deployed to the Garden cluster and the destination seed and reconciled by the backup entry controller. As a result, a secret with the appropriate credentials for accessing the source seed backup container named `source-etcd-backup` is created in the destination seed. The normal backup entry (referencing the destination seed backup container) is also deployed and reconciled, as usual, resulting in the usual `etcd-backup` secret being created.
+
+3. A special "copy" version of the `etcd-main` Etcd resource is deployed to the destination seed. In its `backup` section, this resource contains a `sourceStore` in addition to the usual `store`, which contains the parameters needed to use the source seed backup container, such as its name and the secret created in the previous step.
+
+   ```yaml
+   spec:
+     backup:
+       ...
+       store:
+         container: 408740b8-6491-415e-98e6-76e92e5956ac
+         secretRef:
+           name: etcd-backup
+         ...
+       sourceStore:
+         container: d1435fea-cd5e-4d5b-a198-81f4025454ff
+         secretRef:
+           name: source-etcd-backup
+         ...
+   ```
+
+4. The `etcd-druid` in the destination seed reconciles the above resource by deploying a `etcd-main` StatefulSet that doesn't contain the main `etcd` container at all. Instead, it only contains the `backup-restore` container that runs `etcd-backup-restore` configured to run in "copy" mode, which would only perform copying of the snapshots from the source to the destination backup container.
+   
+5. Before starting the copy itself, the `etcd-backup-restore` process in the destination seed creates the `CopyOperation` sync object mentioned above (if it doesn't already exist; see also [Migration Flow Adaptations](#migration-flow-adaptations)) with its status set to `Initial`. It then waits for the status of this sync object to become `Ready` in order to proceed. This waiting is up to a certain timeout that should be sufficient for a full snapshot to be taken; after this timeout has elapsed, it sets the status to `Ready` on its own, and the reconciliation flow continues from step 8, as it relies on the safety mechanism described in [Implicit Owner Election](#implicit-owner-election) below.
+
+6. The `etcd-backup-restore` process in the source seed detects the `CopyOperation` sync object in status `Initial` and switches to a special "final snapshot" mode. In this mode the regular snapshotter is stopped, the readiness probe of the main `etcd` container starts returning 503, and one final full snapshot is taken. Finally, the status of the `CopyOperation` is set to `Ready`.
```

Comment:
> I missed the details how the final snapshot is taken in the "good" case

See the "Migration Flow Adaptations" section, it's described there in quite some detail: *Instead of taking a full snapshot of the source seed `etcd`, the flow initiates a copy operation by creating a CopyOperation sync object with status Initial in the source backup container, and waits for its status to become Ready, as the last step before deleting the shoot namespace in the source seed. (and so on)*

As a general remark, you could perhaps read the document to the end before starting to comment? 

> Now, how are the "threads of both cases joined"? ...

I would refrain from answering all the questions here. The short version: the 2 processes are synchronizing over the status of the `CopyOperation`. Once it's `Initial`, the source process knows it should take a final snapshot, set it to `Ready`, and stop doing anything else. Once it's `Ready`, the destination process knows it should start copying the snapshots. Normally the `Ready` state is set by the source, but it can be set by the destination if the source didn't update it for too long. In the worst case, the source could have already seen the `Initial` (woken up by some sort of miracle) and started creating the final snapshot, which won't be copied by the destination in this case - but I don't think this is a big deal. All of this is true no matter "good case" or "bad case", since the sync mechanism is one and the same.

> Will the destination seed shoot backup know that it already has a final backup in the blob store, i.e. is a final backup marked as such, so that the destination seed knows it has already the latest (and final) backup?

Not needed, once it finds `Ready` status that it didn't set itself the newest snapshot is the final snapshot. The source process simply won't take any more snapshots, unless the sync object is deleted or changed, which can only happen by a manual intervention.

> Or will the destination seed have to wait until its timeout lapses before it just takes whatever last backup it finds, final or otherwise?

If the timeout elapsed, it will take whatever last backup if finds. The assumption is that the source seed can't even connect to the backup bucket in this case and it could stay like that for quite some time. So we restore whatever we can, rather than waiting forever.

> Or is the good case updated to work already with the CopyOperation (introduced with the "bad" case) and that's how the "final snapshot" will be triggered in both cases ("good" and "bad")?

Yes, once again, see "Migration Flow Adaptations".

Reply:
In the latest proposal, there is no more `CopyOperation`. Special care is still taken to ensure that the "good case" and "bad case" are compatible with each other since they rely on the same basic mechanism (this time is the ownership check via the owner DNS record).